azure-pipelines.yml

```diff
@@ -122,10 +122,30 @@ jobs:
       pythonVersion: '$(pythonVersion)'
       workerPath: '$(workerPath)'
       artifactName: '$(pythonVersion)_OSX_X64'
+- job: Build_OSX_ARM64
+  pool:
+    vmImage: 'macOS-latest'
+  strategy:
+    matrix:
+      Python39V4:
+        pythonVersion: '3.9'
+        workerPath: $(PROD_V4_WORKER_PY)
+      Python310V4:
+        pythonVersion: '3.10'
+        workerPath: $(PROD_V4_WORKER_PY)
+      Python311V4:
+        pythonVersion: '3.11'
+        workerPath: $(PROD_V4_WORKER_PY)
+  steps:
+  - template: pack/templates/macos_64_env_gen.yml
+    parameters:
+      pythonVersion: '$(pythonVersion)'
+      workerPath: '$(workerPath)'
+      artifactName: '$(pythonVersion)_OSX_ARM64'
 
 - job: PackageWorkers
-  dependsOn: ['Build_WINDOWS_X64', 'Build_WINDOWS_X86', 'Build_LINUX_X64', 'Build_OSX_X64']
-  condition: or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+  dependsOn: ['Build_WINDOWS_X64', 'Build_WINDOWS_X86', 'Build_LINUX_X64', 'Build_OSX_X64', 'Build_OSX_ARM64']
+  condition: or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(variables['Build.SourceBranch'], 'refs/heads/dev'), variables['GeneratePackage'])
```

Comment:
Adding a new variable `variables['GeneratePackage']` in the azure pipeline can trigger this job, and there is no need to play with the branch names to get it running.